apps/src/lib/node/ledger/ethereum_oracle/mod.rs

```diff
@@ -53,12 +47,132 @@ pub enum Error {
     Timeout,
 }
 
+/// Convert values to [`ethabi`] Ethereum event logs.
+pub trait IntoEthAbiLog {
+    /// Convert an Ethereum event log to the corresponding
+    /// [`ethabi`] type.
+    fn into_ethabi_log(self) -> ethabi::RawLog;
+}
+
+impl IntoEthAbiLog for ethers::types::Log {
+    #[inline]
+    fn into_ethabi_log(self) -> ethabi::RawLog {
+        self.into()
+    }
+}
+
+impl IntoEthAbiLog for ethabi::RawLog {
+    #[inline]
+    fn into_ethabi_log(self) -> ethabi::RawLog {
```

Comment:
Is there a need to make a custom trait here? Rust automatically generates an `impl From<T> for T`